server/portal/apps/workspace/api/views.py

```diff
@@ -154,6 +145,44 @@ def get(self, request, *args, **kwargs):
             },
             encoder=BaseTapisResultSerializer
         )
+    
+    def listing(self, client, request):
+        limit = int(request.GET.get('limit', 10))
+        offset = int(request.GET.get('offset', 0))
+        portal_name = settings.PORTAL_NAMESPACE
+
+        data = client.jobs.getJobSearchList(
+            limit=limit,
+            startAfter=offset,
+            orderBy='lastUpdated(desc),name(asc)',
+            _tapis_query_parameters={'tags.contains': portal_name}
+        )
+
+        return data
+
+    def search(self, client, request):
+        query_string = request.GET.get('query_string')
+
+        limit = int(request.GET.get('limit', 10))
+        offset = int(request.GET.get('offset', 0))
+        portal_name = settings.PORTAL_NAMESPACE
+
+        data = []
+        for parameter in job_search_parameters:
+            data = client.jobs.getJobSearchList(
+                limit=limit,
+                startAfter=offset,
+                orderBy='lastUpdated(desc),name(asc)',
+                _tapis_query_parameters={
+                    'tags.contains': portal_name,
+                    f'{parameter}.like' : f'*{query_string}*'
+                }
+            )
+
+            if data:
+                break
```

Comment:
I wonder if there's a way to structure this as one query joined by OR statements. Could we ask CIC?

Reply:
Yes, I will inquire with CIC for a potential solution for this